model/patch_lifecycle.go

```diff
@@ -425,9 +426,156 @@ func FinalizePatch(ctx context.Context, p *patch.Patch, requester string, github
+			status:         alias.Status,
+			parentAsModule: alias.ParentAsModule,
+		}
+		aliasGroups[group] = append(aliasGroups[group], alias)
+	}
+
+	childPatchIDs := []string{}
+	for group, definitions := range aliasGroups {
+		if group.status == "" {
+			childPatchID, err := makePatchFromDefinitions(ctx, p, group.project, group.parentAsModule, definitions)
+			if err != nil {
+				return errors.Wrap(err, "problem creating patch for alias definitions")
+			}
+			if childPatchID != "" {
+				childPatchIDs = append(childPatchIDs, childPatchID)
+			}
+		} else {
+			if err := subscribeOnPatchOutcome(p, group.status, definitions); err != nil {
+				return errors.Wrap(err, "problem subscribing alias definitions on patch outcome")
+			}
+
+		}
+	}
+
+	return p.AppendChildPatches(childPatchIDs)
+}
+
+func makePatchFromDefinitions(ctx context.Context, p *patch.Patch, projectID, parentAsModule string, definitions []PatchTriggerDefinition) (string, error) {
+	if len(definitions) == 0 {
+		return "", nil
+	}
+
+	v, project, err := FindLatestVersionWithValidProject(projectID)
+	if err != nil {
+		return "", errors.Wrapf(err, "problem getting last known project for '%s'", projectID)
+	}
+
+	yamlBytes, err := yaml.Marshal(project)
+	if err != nil {
+		return "", errors.Wrap(err, "can't marshal child project")
+	}
+
+	matchingTasks, err := project.VariantTasksForSelectors(definitions, p.GetRequester())
+	if err != nil {
+		return "", errors.Wrap(err, "problem matching tasks to alias definitions")
+	}
+
+	user, err := user.FindOneById(p.Author)
+	if err != nil {
+		return "", errors.Wrapf(err, "can't get user '%s'", p.Author)
+	}
+	if user == nil {
+		return "", errors.New("can't find parent patch author")
+	}
+
+	patchNumber, err := user.IncPatchNumber()
+	if err != nil {
+		return "", errors.Wrapf(err, "can't increment patch number for '%s'", user.Id)
+	}
+
+	newPatchID := mgobson.NewObjectId()
+	patchDoc := &patch.Patch{
+		Id:            newPatchID,
+		Githash:       v.Revision,
+		VariantsTasks: matchingTasks,
+		Project:       projectID,
+		Author:        p.Author,
+		Description:   fmt.Sprintf("Child patch for '%s'", p.Id.Hex()),
+		Status:        evergreen.PatchCreated,
+		CreateTime:    time.Now(),
+		PatchNumber:   patchNumber,
+		PatchedConfig: string(yamlBytes),
+		Triggers:      patch.TriggerInfo{ParentPatch: p.Id.Hex()},
+	}
+
+	if parentAsModule != "" {
+		for _, p := range p.Patches {
+			if p.ModuleName == "" {
+				patchDoc.Patches = append(patchDoc.Patches, patch.ModulePatch{
+					ModuleName: parentAsModule,
+					PatchSet:   p.PatchSet,
+				})
+				break
+			}
+		}
+	}
+
+	if err = patchDoc.Insert(); err != nil {
+		return "", errors.Wrap(err, "can't insert child patch")
+	}
+
+	token, err := evergreen.GetEnvironment().Settings().GetGithubOauthToken()
+	if err != nil {
+		return "", errors.Wrap(err, "error getting github token from settings")
+	}
+
+	_, err = FinalizePatch(ctx, patchDoc, patchDoc.GetRequester(), token)
```

Comment:
I believe this makes finalizing a patch that has downstream triggers defined actually finalize 2 patches, and finalizing 1 can already be slow. Would it be better to have a job do this similar to the intent processor, or do we need this to be synchronous?